lib/cell.rb

```diff
@@ -0,0 +1,48 @@
+require './lib/ship'
+
+class Cell
+
+  attr_accessor :coords, :ship_position, :coordinate
+  attr_accessor :ship, :miss
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = ship
+    @ship_position = ""
+  end
+
+  def place_ship(ship)
+    @ship = ship
+    ship_position = @coordinate
+    puts
+  end
+
+  def empty?
+    ship_position == ""
+  end
+
+  def fired_upon?
+    @ship.health < @ship.length
+  end
+
+  def fire_upon
+    if @coordinate = ship_position
+      @ship.hit
+    else
+      @miss = true
+  end
+
+  def render(reveal = false)
+    if reveal == true
+      puts "S"
+    end
+
+    if fired_upon? == true
+      ship_position = "H"
+    elsif fired_upon? == true && @ship.health = 0
```

Comment:
Wondering if this line will needs to be a loop, since it will need to update all of the H's to X's